common/domain/replication_queue_test.go

```diff
@@ -619,3 +710,28 @@ func TestPurgeAckedMessages(t *testing.T) {
 		})
 	}
 }
+
+func TestReplicationQueueImpl_purgeProcessor(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockQueue := persistence.NewMockQueueManager(ctrl)
+	rq := NewReplicationQueue(mockQueue, "testCluster", nil, nil).(*replicationQueueImpl)
+	atomic.StoreInt32(&rq.status, common.DaemonStatusStarted)
+
+	done := make(chan bool)
+	mockQueue.EXPECT().GetAckLevels(gomock.Any()).Return(map[string]int64{}, nil).AnyTimes()
+	mockQueue.EXPECT().DeleteMessagesBefore(gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+
+	go func() {
+		rq.purgeProcessor()
```

Comment:
instead of calling `purgeProcessor` directly you can start the replication queue. it will give us better coverage by validating that start eventually calls purgeProcessor

Reply:
that is already done in testing for Start. This is specically for purge, do you want me to remove it then?